proxy/proxy.go

```diff
@@ -51,7 +60,8 @@ func Decompress(rdr io.ReadCloser, encoding string) (io.ReadCloser, error) {
 		if err != nil {
 			return nil, err
 		}
-		return io.NopCloser(decomp), nil
+		// else we leak
+		return &zstdCloser{Decoder: decomp}, nil
```

Comment:
Why not `return decomp, nil`?

Reply:
because the signature is wrong, decomp close returns `Close()` and not `Close() error`